File: OOP/5_multiple_inheritance.py
class Father:
    def __init__(self,father_name) -> None:
        self.father_name = father_name
    def show_father(self):
        return f"Fathers' name: {self.father_name}"

class Mother:
    def __init__(self,mother_name) -> None:
        self.mother_name = mother_name
    def show_mother(self):
        return f"Mothers' name: {self.mother_name}"

class Child(Father,Mother):
    def __init__(self,father_name,mother_name,child_name) -> None:
        Father.__init__(self,father_name)
        Mother.__init__(self,mother_name)
        self.child_name = child_name
    
    def show_child(self):
        return f"Childs' name: {self.child_name}"
    
# as the child class inherited from multiple classes,it represents multiple inheritance
obj1 = Child("John","Stephany","Asif")
print(obj1.show_father())
print(obj1.show_mother())
print(obj1.show_child())
        

File: OOP/5_multiple_inheritance.cpp
#include <iostream>
#include <string>
using namespace std;

class Father{
    protected:
        string father_name;
    public:
        Father(string father_name){
            this->father_name = father_name;
        }

        string show_father(){
            return "Fathers' name: " + father_name;
        }
};

class Mother{
    protected:
        string mother_name;
    public:
        Mother(string mother_name){
            this->mother_name = mother_name;
        }
        string show_mother(){
            return "Mothers' name: " + mother_name;
        };
};

class Child: public Father,public Mother{
    private:
        string child_name;
    public:
        Child(string father_name,string mother_name,string child_name):
        Father(father_name),Mother(mother_name){
            this->child_name = child_name;
        }

        string show_child(){
            return "Childs' name: " + child_name;
        }
};

int main(){
    Child obj1("John","Stephany","Asif"); 
    
    cout<< obj1.show_child() << endl;
    cout<< obj1.show_father() << endl;
    cout<< obj1.show_mother() << endl;
}